src/screens/SignupScreen.js

```diff
@@ -48,71 +49,79 @@ const SignupScreen = ({ navigation }) => {
 
   async function handleSubmit() {
     console.log("clicked");
+    setIsLoading(true);
     const isUsernameNotAvailable = await isUsernameTaken(credentials.username);
 
     if (!isValidUsername(credentials.username)) {
       Alert.alert(
         "invalid username",
         "username must only contains alphanumeric characters and must at least be 1 character long"
       );
+      setIsLoading(false);
     } else if (isUsernameNotAvailable) {
       Alert.alert("username is not available");
+      setIsLoading(false);
     } else if (!isValidEmail(credentials.email)) {
       Alert.alert("email is not valid!");
+      setIsLoading(false);
     } else if (credentials.password !== credentials.confirmPass) {
       Alert.alert("password doesn't match");
+      setIsLoading(false);
     } else if (isPasswordTooShort(credentials.password)) {
       Alert.alert("password must at least be 6 characters long");
+      setIsLoading(false);
     } else {
-      signUp(credentials, () => redirectToLoginScreen(navigation));
+      await signUp(credentials, () => redirectToLoginScreen(navigation));
```

Comment:
I think need to add `setIsLoading(false)` after this. (If email is already taken, the loading animation won't stop)